examples/permissioned-token/test/tests.js

```diff
@@ -0,0 +1,334 @@
+        sign: types.SignType.SecretKey,
+        fromAccount: from.account,
+        appId: permissionsAppId,
+        payFlags: { totalFee: 1000 },
+        appArgs: ['str:transfer'],
+        accounts: [to.address]
+      }
+    ];
+    runtime.executeTx(txGroup);
+    syncInfo();
+  }
+
+  this.beforeEach(async function () {
+    // Create Accounts and Env
+    alice = new AccountStore(minBalance, { addr: ALICE_ADDRESS, sk: new Uint8Array(0) });
+    bob = new AccountStore(minBalance);
+    elon = new AccountStore(minBalance);
+    runtime = new Runtime([master, alice, bob, elon]);
+
+    // Deploy ASA
+    assetIndex = runtime.addAsset('gold', { creator: { ...alice.account, name: 'alice' } });
+    asaDef = runtime.getAssetDef(assetIndex);
+
+    // Setup Controller SSC
+    let sscFlags = {
+      sender: alice.account,
+      localInts: 0,
+      localBytes: 1,
+      globalInts: 4,
+      globalBytes: 2,
+      appArgs: [`int:${assetIndex}`],
+      foreignAssets: [assetIndex]
+    };
+    CONTROLLER_PROGRAM = getProgram(CONTROLLER, { TOKEN_ID: assetIndex });
+    controllerAppID = runtime.addApp(
+      sscFlags, {}, CONTROLLER_PROGRAM, CLEAR_STATE_PROGRAM
+    );
+
+    // Setup Permissions SSC
+    PERMISSIONS_PROGRAM = getProgram(PERMISSIONS, { CONTROLLER_APP_ID: controllerAppID });
+    sscFlags = {
+      sender: alice.account,
+      localInts: 1,
+      localBytes: 0,
+      globalInts: 3,
+      globalBytes: 1,
+      appArgs: [`int:${controllerAppID}`]
+    };
+    permissionsAppId = runtime.addApp(
+      sscFlags, {}, PERMISSIONS_PROGRAM, CLEAR_STATE_PROGRAM
+    );
+
+    // Add permissions SSC config to Controller SSC
+    const appArgs = [
+      'str:add_permission',
+      `int:${permissionsAppId}`,
+      `addr:${alice.address}`
+    ];
+    runtime.executeTx({
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: alice.account,
+      appId: controllerAppID,
+      payFlags: { totalFee: 1000 },
+      appArgs: appArgs,
+      foreignAssets: [assetIndex]
+    });
+
+    // Deploy Clawback Lsig and Modify Asset
+    CLAWBACK_PROGRAM = getProgram(CLAWBACK, {
+      TOKEN_ID: assetIndex,
+      CONTROLLER_APP_ID: controllerAppID
+    });
+    lsig = runtime.getLogicSig(CLAWBACK_PROGRAM, []);
+    clawbackAddress = lsig.address();
+    fund(clawbackAddress);
+    runtime.executeTx({
+      type: types.TransactionType.ModifyAsset,
+      sign: types.SignType.SecretKey,
+      fromAccount: alice.account,
+      assetID: assetIndex,
+      fields: {
+        manager: asaDef.manager,
+        reserve: asaDef.reserve,
+        freeze: asaDef.freeze,
+        clawback: clawbackAddress
+      },
+      payFlags: { totalFee: 1000 }
+    });
+    runtime.optIntoASA(assetIndex, clawbackAddress, {});
+
+    // Refresh Accounts
+    syncInfo();
+  });
+
+  it('Token Issuance', () => {
+    // Cannot issue before opting-in
+    assert.throws(() => issue(elon, 20), 'RUNTIME_ERR1404');
```

Comment:
Please add tests in two flows, one with passing flow(with correct calls, etc), other one is failing paths(which will have corner cases, failing cases).

Reply:
Are you referring to 2 separate files?